alacritty/src/display/hint.rs

```diff
@@ -259,30 +342,65 @@ pub fn highlighted_at<T>(
             return None;
         }
 
-        hint.regex.with_compiled(|regex| {
-            // Setup search boundaries.
-            let mut start = term.line_search_left(point);
-            start.line = max(start.line, point.line - MAX_SEARCH_LINES);
-            let mut end = term.line_search_right(point);
-            end.line = min(end.line, point.line + MAX_SEARCH_LINES);
+        // TODO: Use `bool::then` instead of double-if when MSRV >= 1.50.0
+        if hint.hyperlinks {
+            if let Some((hyperlink, bounds)) = hyperlink_at(term, point) {
+                return Some(HintMatch::Hyperlink {
+                    hyperlink,
+                    bounds,
+                    action: hint.action.clone(),
+                });
+            }
+        }
 
-            // Function to verify that the specified point is inside the match.
-            let at_point = |rm: &Match| *rm.end() >= point && *rm.start() <= point;
+        if let Some(bounds) = hint.regex.as_ref().and_then(|regex| {
+            regex.with_compiled(|regex| regex_match_at(term, point, regex, hint.post_processing))
+        }) {
+            return Some(HintMatch::Regex { bounds, action: hint.action.clone() });
+        }
+
+        None
+    })
+}
 
-            // Check if there's any match at the specified point.
-            let mut iter = RegexIter::new(start, end, Direction::Right, term, regex);
-            let regex_match = iter.find(at_point)?;
+/// Retrive the hyperlink with its range, if there is one at the specified point.
+fn hyperlink_at<T>(term: &Term<T>, point: Point) -> Option<(Hyperlink, Match)> {
+    let _ = term.hyperlink_at(point)?;
```

Comment:
```suggestion
    term.hyperlink_at(point)?;
```

I don't think it should be necessary to bind to `_` here?

Also the logic in this method seems extremely strange to me. First we check the terminal if there's just a simple hyperlink, it tells us there is, then we go on and iterate over **all** visible hyperlinks to find it. Seems inefficient?